ocaml/xapi/test_pusb.ml

```diff
@@ -0,0 +1,78 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open OUnit
+open Test_common
+
+let create_base_environment () =
+  let __context = make_test_database () in
+  let pusb = make_sr ~__context () in
+  __context, pusb
+
+let test_scan ~__context info =
+  let usbs = Xapi_pusb_helpers.get_usbs info in
+  let f () = Xapi_pusb.scan_start ~__context usbs in
+  Xapi_pusb.start_thread f;
+  Xapi_pusb.scan ~__context;
+  Thread.delay 1.0;
```

Comment:
What's the purpose of this delay?

Reply:
This delay is to delay 1 second after first scan call to make sure the PUSB records is updated.(assume the scan may take a little long time).

Reply:
For unit test, I have refined. Should review it again. Thansk @jonludlam